From e154cae59199583eac817d53fb5aed5724a8f1b7 Mon Sep 17 00:00:00 2001
From: Shen Zhenyi <szy@rock-chips.com>
Date: Mon, 20 Jun 2016 11:35:42 +0800
Subject: [PATCH] video: rockchip: hdmi: yuv420 resolution retain 4K 50/60HZ

Change-Id: I75ae87bbd274af10b7da9b6699d5892e6f864dba
Signed-off-by: Shen Zhenyi <szy@rock-chips.com>
(cherry picked from commit 6acb6585c03dd4bf7eaf2a548cb3f836070ba56c)
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c
index 1d3cce5a3fd2..9b0b945081c5 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-lcdc.c
@@ -1126,6 +1126,18 @@ int hdmi_add_vic(int vic, struct list_head *head)
 	if (vic == 0)
 		return -1;
 
+	if (vic & HDMI_VIDEO_YUV420) {
+		v = vic & 0xff;
+		if (v != HDMI_3840X2160P_50HZ &&
+		    v != HDMI_3840X2160P_60HZ &&
+		    v != HDMI_4096X2160P_50HZ &&
+		    v != HDMI_4096X2160P_60HZ &&
+		    v != HDMI_3840X2160P_50HZ_21_9 &&
+		    v != HDMI_3840X2160P_60HZ_21_9) {
+			return -1;
+		}
+	}
+
 	list_for_each(pos, head) {
 		modelist = list_entry(pos, struct display_modelist, list);
 		v = modelist->vic;
-- 
2.35.3

